controllers/user.go

```diff
@@ -79,7 +59,18 @@ func (i implement) DeleteUser(userID string) (bool, error) {
 	return false, nil
 }
 
-func (i implement) UpdateUser(user *models.User) (models.User, error) {
-	//TODO implement me
-	panic("implement me")
+func (i implement) UpdateUserByID(user *models.User) (models.User, error) {
+	// if user not found return err, check by id
+	for _, usr := range users {
+		if usr.ID == user.ID {
+			usr.Username = user.Username
+			usr.Password = user.Password
+			return models.User{
+				ID:       user.ID,
+				Username: user.Username,
+				Password: user.Password,
```

Comment:
có thể tìm user bằng user id khi để trong loop for. nhưng câu if dòng 65 có thể để ngoài vòng loop

Reply:
controller tương tác tới DB? sao k thấy sử dụng?

Reply:
connect với DB